lib/beaker/host/eos.rb

```diff
@@ -0,0 +1,56 @@
+[ 'host', 'command_factory' ].each do |lib|
+  require "beaker/#{lib}"
+end
+
+module Eos
+  class Host < Unix::Host
+
+    # Gets the path & file name for the puppet agent package on EOS
+    #
+    # @param [String] puppet_collection Name of the puppet collection to use
+    # @param [String] puppet_agent_version Version of puppet agent to get
+    # @param [Hash{Symbol=>String}] opts Options hash to provide extra values
+    #
+    # @raise [ArgumentError] If one of the two required parameters (puppet_collection,
+    #   puppet_agent_version) is either not passed or set to nil
+    #
+    # @return [String, String] Path to the directory and filename of the package, respectively
+    def get_puppet_agent_package_info( puppet_collection = nil, puppet_agent_version = nil, opts = {} )
+      error_message = "Must provide %s argument to get puppet agent package information"
+      raise ArgumentError, error_message % "puppet_collection" unless puppet_collection
+      raise ArgumentError, error_message % "puppet_agent_version" unless puppet_agent_version
+
+      variant, version, arch, _ = self['platform'].to_array
+      release_path = "#{variant}/#{version}/#{puppet_collection}/#{arch}"
+      release_file = "puppet-agent-#{puppet_agent_version}-1.#{variant}#{version}.#{arch}.swix"
```

Comment:
To be honest, I don't know, I just know that this works.  You'd have to ask RE for that info.  My first suspicion would have been puppet collection, but we already take that into account right above it.